File: py/merger.py

first_names = ["Aaron", "Tom", "Brock", "Gyallis", "Rick"]
last_names = ["Rodgers", "Brady", "Lesnar", "Banton", "Ross"]


def merge2Single (arr1,arr2):
    words = []  
    for i in range (len(arr1)):
        words.append (arr1[i] +" "+arr2[i])
    return words

def merge2Object (arr1,arr2):
    # names = {"firstname": first_names, "lastname":last_names}
    # for i in range (len(arr1)):
    #     names.update ({arr1[i] : arr2[i]})
    d = {}
    names = []
    for i in range (0,len(arr1)):
        d = {'firstname':arr1[i],'lastname':arr2[i]}
        names.append(d)

    
    
    return(names)

     
    


def mergeHandler (func,arr1,arr2):
    return (func(arr1, arr2))


print (mergeHandler(merge2Single, first_names,last_names))
print (mergeHandler(merge2Object,first_names,last_names))

File: js/merger.js
var first_names = ["ashida","tom","shelly","kyle","nicholas"];
var last_names = ["ali","brady","ann","defreitas","mendez"];

function mergeHandler(arr1,arr2,func){
    return (func(arr1,arr2))

}
 

function merge2Single(arr1,arr2){
  let words =[];
    for( i=0; i < arr1.length; i++){
        words.push (arr1[i] +" "+ arr2[i] );
    }
  return words;
}
function merge2Object(arr1,arr2){
  d = {};
  names = [];
   for (i=0; i < arr1.length;i++){
     d = {'firstname':arr1[i],'lastname':arr2[i]}
        names.push(d)
   }
    return names;
      

}
console.log(mergeHandler(first_names,last_names,merge2Single));
console.log(mergeHandler(first_names,last_names,merge2Object));


module.exports.mergeHandler= mergeHandler;
module.exports.merge2Single= merge2Single;
module.exports.merge2Object = merge2Object;
